backend/codingchallenge/api/views/admin/adminApplicationView.py

```diff
@@ -0,0 +1,104 @@
+from rest_framework.permissions import IsAdminUser, IsAuthenticated
+from django.contrib.auth.models import User
+
+# RESTapi imports
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+# import model
+from ...models import Application, Challenge
+
+# import serializer
+from ...serializers import ApplicationSerializer
+
+
+#
+
+### endpoint: /api/admin/applications
+class AdminApplicationView(APIView):
+    # grant permission only for admin user
+    permission_classes = [IsAdminUser]
+
+    name = "Admin Application View"
+    description = "handling all requests for applications as a admin"
+
+    # 1. List all
+    def get(self, request, *args, **kwargs):
+        '''
+        get Application with follwing columns:
+            applicationId,
+            operatingSystem,
+            programingLanguage,
+            expiry,
+            submission,
+            githubRepo,
+            status,
+            created,
+            modified
+        '''
+        applications = Application.objects
+        serializer = ApplicationSerializer(applications, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    # 2. Create
+    def post(self, request, *args, **kwargs):
+        '''
+        post Application with
+            required arguments:
+                applicationId,
+                applicantEmail
+
+            optional arguments:
+                challengeId
+                days
+        '''
+
+        try:
+            # random challenge
+            challengeId = random.choice(Challenge.objects.all()).id
+            # default 2 days time for start
+            days = 2
+
+        except IndexError:
+            return Response({'detail': 'there are no challenges in database'},
+                            status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+        if len(request.data) == 0:
+            return Response({'detail': 'body is empty'}, status=status.HTTP_204_NO_CONTENT)
+
+        try:
+
+            if Application.objects.all().filter(
+                    applicationId=request.data.get('application').get('applicationId')):
+                return Response({'detail': 'applicationId already in use'}, status=status.HTTP_409_CONFLICT)
+
+            if 'challengeId' in request.data.get('application'):
+                challengeId = request.data.get('application').get('challengeId')
+
+            if 'days' in request.data.get('application'):
+                days = request.data.get('application').get('days')
+
+        except AttributeError:
+            return Response({'detail': 'wrong json attributes'}, status=status.HTTP_400_BAD_REQUEST)
+
+        data = {
+            'applicationId': request.data.get('application').get('applicationId'),
+            'applicantEmail': request.data.get('application').get('applicantEmail'),
+            'challengeId': challengeId,
+            'expiry': time.time() + days * 24 * 60 * 60
```

Comment:
int(time.time()) is better to avoid any conversions to string. Otherwise it could be possible to get something like that: Tue Apr 18 21:35:10 2023

Reply:
Cast is not necessary, because in database any unix timestamp is stored as real.
So the timestamps can be more exact (float)